proto/spire/plugin/agent/nodeattestor/v1/nodeattestor.proto

```diff
@@ -1,37 +1,50 @@
-/** Responsible for attesting the physical nodes identity.  The plugin
-will be responsible to retrieve an identity document or data associated
-with the physical node.  This data will be used when attesting to the server.
-*/
-
 syntax = "proto3";
-package spire.agent.nodeattestor;
-option go_package = "github.com/spiffe/spire/proto/spire/agent/nodeattestor";
+package spire.plugin.agent.nodeattestor.v1;
+option go_package = "github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/agent/nodeattestor/v1;nodeattestorv1";
 
-import "spire/common/common.proto";
-import "spire/common/plugin/plugin.proto";
+service NodeAttestor {
+    // FetchAttestationData retrieves the attestation payload and optionally
+    // participates in challenge/response attestation mechanics.
+    //
+    // The attestation flow is as follows:
+    // 1. SPIRE Agent opens up a stream to the plugin via FetchAttestationData.
+    // 2. The plugin returns a response with the payload.
+    // 3. SPIRE Agent sends the payload to SPIRE Server.
+    // 4. Optionally, SPIRE Server responds with a challenge:
+    //   4a. SPIRE Agent sends the challenge to the plugin.
+    //   4b. The plugin responds with the challenge response.
+    //   4c. SPIRE Agent sends the challenge response to SPIRE Server.
+    //   4d. Step 4 is repeated until SPIRE Server is satisfied and does not
+    //       respond with an additional challenge.
+    // 5. SPIRE Agent closes the stream.
+    //
+    // Note that SPIRE Agent does NOT send a request down the stream unless it
+    // needs to issue the challenge returned by SPIRE Server (step 4a).
+    //
+    // Plugins that do not need challenge/response as part of the attestation
+    // process may close the stream as soon as they send the attestation
+    // payload (step 2).
+    rpc FetchAttestationData(stream FetchAttestationDataRequest) returns (stream FetchAttestationDataResponse);
+}
 
-/** Represents an empty request */
 message FetchAttestationDataRequest {
```

Comment:
Another naming possibility is this could be `Challenge`, the other could be `AttestResponse`